src/app/sitemap.ts

```diff
@@ -0,0 +1,85 @@
+import type { MetadataRoute } from "next";
+import { getBlogPosts } from "@/services/api";
+import { cookies } from "next/headers";
+
+export default async function sitemap(): Promise<MetadataRoute.Sitemap> {
+  cookies()
+  const now = Date.now();
+  // Filter non-archived blog posts
+  const blogPosts: MetadataRoute.Sitemap = (await getBlogPosts())
+    .filter((b) => {
+      if (b.end_at !== null && b.end_at !== undefined) {
```

Comment:
unfortunately the 'nullable' OpenAPI option doesn't tell whether it would be `undefined` or `null` since both will be valid, so the type generated for nullable will always be `string | null | undefined`.

***

There is a tslint rule `@typescript-eslint/strict-boolean-expressions` that keep bothering me to always use strictly boolean expression in conditional statements, otherwise I would write sth like: 
```js
if (b.end_at) { ... }
```
I might actually turn that rule off in the future as it's simply not needed for our use cases.